tests/test_jsvalidation.py

```diff
@@ -0,0 +1,597 @@
+                    "field_B": {"type": "integer"}
+                }
+            }
+        ]
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+
+
+def test_conflict_in_collection():
+    """Test that having a conflict between elements of a collection fails."""
+    v1 = {
+        "type": "object",
+        "oneOf": [
+            {
+                "type": "object",
+                "properties": {
+                    "field_A": {"type": "string"}
+                }
+            }, {
+                "type": "object",
+                "properties": {
+                    "field_A": {"type": "integer"}
+                }
+            }
+        ]
+    }
+    obj = JSONSchemaValidator()
+    with pytest.raises(JSONSchemaCompatibilityError):
+        obj.validate(v1, 'first')
+
+
+def test_conflict_in_collection2():
+    """Test that having a conflict between elements of a collection fails."""
+    v1 = {
+        "type": "object",
+        "allOf": [
+            {
+                "type": "object",
+                "properties": {
+                    "my_field": {
+                        "type": "string"
+                    }
+                }
+            },
+            {
+                "type": "object",
+                "dependencies": {
+                    "my_field": {
+                        "type": "object",
+                        "properties": {
+                            "my_field": {
+                                "type": "integer"
+                            }
+                        }
+                    }
+                }
+            }
+        ]
+    }
+    obj = JSONSchemaValidator()
+    with pytest.raises(JSONSchemaCompatibilityError):
+        obj.validate(v1, 'first')
+
+
+def test_ref_pass():
+    """Test that having no conflict in schema with references passes."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"},
+                    "city": {"type": "string"},
+                    "state": {"type": "string"}
+                },
+                "required": ["street_address", "city", "state"]
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/address"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        },
+                        "required": ["type"]
+                    }
+                ]
+            }
+        }
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+
+
+def test_ref_no_conflict_inside_schema_pass():
```

Comment:
This test and the previous one seem to be exactly the same. We should remove one.